doc/HOWTO/howto_setupAndDeployCustom.md

```diff
@@ -0,0 +1,144 @@
+# HOWTO: Set Up a Testnet and Deploy Custom Contracts
+
+## Goals 
+This HOWTO details the steps required to set up a local Stellar network, deploy a custom Soroban contract, and invoke the contract via CLI. It complements the [official guide][guide].
+
+## Process
+
+Prerequisites:
+- Make `cargo`, `rustup`, and `go` available. You will need a`rustc` version that is at least 1.74.
+- Make sure you have enough allocated space. If you're using the NixOS VM, you might need to add `services.logind.extraConfig = "RuntimeDirectorySize=2G";` to your configuration, rebuild and reboot the VM.
+- you probably want to add `~/.cargo/bin` to your PATH
+
+### Local Stellar Network
+Instead of building from source, Stellar provides a quickstart Docker image. You can call the below command from anywhere, and Docker will fetch the image.
```

Comment:
Since you mention Nixos above, we should add a link to setting up docker on Nixos: https://nixos.wiki/wiki/Docker